yaksh/models.py

```diff
@@ -1164,13 +1180,27 @@ class AnswerPaper(models.Model):
         default='inprogress'
     )
 
+    # set question order
+    questions_order = models.CharField(max_length=255, blank=True, default='')
+
     objects = AnswerPaperManager()
 
     def current_question(self):
         """Returns the current active question to display."""
         if self.questions_unanswered.all():
-            return self.questions_unanswered.all()[0]
-        return self.questions.all()[0]
+            cur_question = self.get_current_question(
+                self.questions_unanswered.all())
+        else:
+            cur_question = self.get_current_question(self.questions.all())
+        return cur_question
+
+    def get_current_question(self, questions):
+        if self.questions_order:
```

Comment:
Why is this if case needed at all, don't you always have a questions_order when you make the paper or is there a case when this doesn't happen?  Maybe there is I am just asking.

Reply:
If this condition is not added then older answer papers where there are no question order will not be able to get questions.